core/dbt/exceptions.py

```diff
@@ -772,6 +772,10 @@ def raise_duplicate_resource_name(node_1, node_2):
     elif node_1.resource_type == NodeType.Documentation:
         get_func = 'doc("{}")'.format(duped_name)
     elif node_1.resource_type == NodeType.Test and 'schema' in node_1.tags:
+        # TODO: This causes duplicate schema test errors to be ignored!!
+        # This PR should ensure this never gets called when the tests are actually different,
+        # but removing this breaks 50+ tests that currently have the exact same schema
+        # test defined more than once. :/
```

Comment:
Anyone else know why this is the way it is?  Git blame isn't helping much.

Reply:
I don't have anything to add to _why_ this exists, but wanted to shine light on something I noticed.

The weird part is that when you remove that short circuiting logic, **source** tests defined once raise a duplicate resource error. At first glance, it looks like we add source tests twice:

https://github.com/fishtown-analytics/dbt/blob/8d39ef16b67fcc3c30e8a18751b4e64831ec9aaf/core/dbt/parser/sources.py#L128-L132

https://github.com/fishtown-analytics/dbt/blob/8d39ef16b67fcc3c30e8a18751b4e64831ec9aaf/core/dbt/parser/schemas.py#L530-L533

I think our tests are doing their job in this case! We should reconcile the duplicate logic adding schema tests on sources, I think this is residual code from swapping parse results with directly using the manifest (#3219)

Reply:
That's pretty strange and not something I've looked at before. As far as I know, it's not related to using the manifest directly, because the sources and nodes were just copied into the manifest and then things proceeded and the copy step just vanished. But I could be wrong.

Reply:
> Do you remember which ones in particular were breaking?
@jtcohen6 I do not, but it's easy enough to generate one.  I remember thinking at the time I discovered it that, whatever is going on there was too large to include as part of this ticket.

@kwigley Ohh interesting about the source tests!  I'll do some digging and see what's happening there.  It's possible those are the test fails I was seeing.

Reply:
> > Do you remember which ones in particular were breaking?
> > @jtcohen6 I do not, but it's easy enough to generate one.  I remember thinking at the time I discovered it that, whatever is going on there was too large to include as part of this ticket.
> 
> @kwigley Ohh interesting about the source tests! I'll do some digging and see what's happening there. It's possible those are the test fails I was seeing.

Awesome! I'm going to mark the review as "Request Changes" just so I know when to take another pass if you come back with anything new after digging in 👍

Reply:
Digging in it appears that @kwigley nailed it.  All the errors were due to source tests being added to the manifest x2.
It seems out of scope here, so I made a [bug ticket](https://github.com/fishtown-analytics/dbt/issues/3347) for it.